### PR TITLE
Fix function related BBEs

### PR DESCRIPTION
**Pull request by pubudu91**

### CODE REVIEW

File: examples/function-pointers/function_pointers.bal [hunk 1]
```diff
@@ -1,41 +1,44 @@
...
 
-// The `test` function acts as a variable function pointer in the `main` function.
-function test(string s, int... x) returns float {
-    int|error y = 'int:fromString(s);
-    float f = 0.0;
+public function main() {
+    // The function name `test` serves as a function pointer argument in the
+    // call to `foo()`. Function names can be thought of as final variables
+    // since although you can use it like a regular variable, you cannot
+    // modify the value it's associated with.
```
Comment (MaryamZi) [inline examples/function-pointers/function_pointers.bal:8]:
```suggestion
    // modify the value it is associated with.
```

Comment (MaryamZi) [reply examples/function-pointers/function_pointers.bal:8]: Do we need to say this specifically though?

File: examples/function-pointers/function_pointers.description [hunk 2]
```diff
@@ -1,4 +1,5 @@
-// A function pointer is a Ballerina type that allows you to use functions as variables, 
-// arguments to functions, and function return values. The name of a function serves as 
-// a pointer to that function when called from other functions or operations. The definition 
-// of the function name provides the type of the pointer in terms of a function signature.
+// Ballerina allows the user to define variables of function types. This allows you to use
```
Comment (KRVPerera) [inline examples/function-pointers/function_pointers.description:1]: I think we start by saying `user` then the second sentence we say `you`. Maybe we can omit `you`?

Comment (MaryamZi) [reply examples/function-pointers/function_pointers.description:1]: Or 
```suggestion
// Ballerina allows defining variables of function types. This allows you to use
```

File: examples/function-pointers/function_pointers.bal (hunk 1 above)
Comment (pubudu91) [reply examples/function-pointers/function_pointers.bal:8]: Added for the sake of completeness, since we're drawing parallels with variables.

**Pull request merged**